test/infrastructure/docker/exp/docker/nodepool.go

```diff
@@ -252,9 +266,12 @@ func (np *NodePool) reconcileMachine(ctx context.Context, machine *docker.Machin
 			return errors.Wrapf(err, "failed to exec DockerMachinePool instance bootstrap for instance named %s", machine.Name())
 		}
 		machineStatus.Bootstrapped = true
+		// return to surface the machine has been bootstrapped.
+		return nil
```

Comment:
I'm confused why you want to return early here. Why would you not want to finish provisioning the machine?

Reply:
Same as above. It is just a way to provide feedback without blocking